### PR TITLE
Loosen curl/libcurl pins to 7

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
Following a discussion in https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/537, it became clear that we don't want to keep updating dependencies for older versions of curl/libcurl simply because the pins haven't been updated.  Migration of certain packages are stuck because of this issue (e.g. https://github.com/conda-forge/nco-feedstock/pull/108).

**Comment by xylar** [comment]
This is also related to https://github.com/conda-forge/curl-feedstock/issues/68

File: recipe/migrations/curl7691.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# To learn more about migrations read CFEP-09
+# https://github.com/conda-forge/conda-forge-enhancement-proposals/blob/master/cfep-09.md
+# The timestamp of when the migration was made
+# Can be obtained by copying the output of
+# python -c "import time; print(f'{time.time():.0f}')"
+migrator_ts: 1586520931
```
Comment (xylar) [inline recipe/migrations/curl7691.yaml:6]: I'm happy to make the `migrator_ts` more precise (as I have seen in some recent migrators) but the comment in the example should probably be updated to show how this is done.

**Comment by xylar** [comment]
I don't have much experience with migrators so I'm not sure how version and build bumping gets handled.  Anyone from @conda-forge/core should feel free to push whatever is needed.

File: recipe/migrations/curl7691.yaml (hunk 1 above)
Comment (CJ-Wright) [reply recipe/migrations/curl7691.yaml:6]: Its fine, the bot uses the raw timestamp from python when it creates these, but it doesn't matter.

**Comment by CJ-Wright** [comment]
Currently curl and libcurl are pinned down only to the major version number so I'm a bit confused here. @isuruf what do you think?

**Comment by xylar** [comment]
@CJ-Wright, I have to admit that I'm also confused.  `curl` and `libcurl` were rebuilt with the latest migration of `krb5`: https://github.com/conda-forge/curl-feedstock/pull/65.  Yet, when trying to migrate `nco` to the new pin of `kbr5`, no compatible host environment can be created, which I traced to the pin to `7.64` rather than just `7`.  That is, there is no `curl` 7.64 that has been built with the new pinned version of `kbr5`.  @ocefpaf then rebuilt `curl` 7.64.0 with the necessary `kbr5` but it became clear that the dependency constraints on packages such as `libnetcdf` was `curl >=7.64.1,<8.0a0`, meaning that rebuilding 7.64.0 wasn't helpful.

My impression from @isuruf was that it made more sense to update the pins to the latest `curl` than it does to try to redundantly update dependencies for the latest `curl` and the pinned `curl`, given that this seems to be necessary.

Hopefully, someone can explain to me at some point what it means for a package to have major and minor version pins but only major version constraints in `pin_run_as_build`, because this seems to relate somehow to the trouble we're having.

**Comment by isuruf** [comment]
> Currently curl and libcurl are pinned down only to the major version number so I'm a bit confused here. @isuruf what do you think?

We don't need a migrator. Just change [this line](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L442) to 7

**Comment by CJ-Wright** [comment]
(also the curl line needs an update to 7)

**Comment by xylar** [comment]
> We don't need a migrator. Just change this line to 7

Okay, I had understand you got pushback on that the last time.  But I agree that solution makes more sense to me, too.

**Comment by isuruf** [comment]
Ping @conda-forge/core

**Comment by beckermr** [comment]
Merge away!

**Pull request merged**